pkg/cvo/status.go

```diff
@@ -293,8 +293,37 @@ func updateClusterVersionStatus(cvStatus *configv1.ClusterVersionStatus, status
 		})
 	}
 
-	progressReason, progressMessage, skipFailure := convertErrorToProgressing(cvStatus.History, now.Time, status)
+	failure := status.Failure
+	var skipFailure bool
+	var progressReason, progressMessage string
+	if !status.Reconciling && len(cvStatus.History) != 0 {
+		progressReason, progressMessage, skipFailure = convertErrorToProgressing(now.Time, failure)
+		failure = filterErrorForFailingCondition(failure, payload.UpdateEffectNone)
+		failureReason, failureMessage := getReasonMessageFromError(failure)
+		statusFailureReason, statusFailureMessage := getReasonMessageFromError(status.Failure)
+		if failureReason != statusFailureReason {
+			klog.Infof("Filtered failure reason changed from '%s' to '%s'", statusFailureReason, failureReason)
+		}
+		if failureMessage != statusFailureMessage {
+			klog.Infof("Filtered failure message changed from '%s' to '%s'", statusFailureMessage, failureMessage)
+		}
+	}
 
+	// set the failing condition
+	failingCondition := configv1.ClusterOperatorStatusCondition{
+		Type:               ClusterStatusFailing,
+		Status:             configv1.ConditionFalse,
+		LastTransitionTime: now,
+	}
+	if failure != nil && !skipFailure {
+		failureReason, failureMessage := getReasonMessageFromError(failure)
```

Comment:
Could we do it like the following?

// right after "var progressReason, progressMessage string"
statusFailureReason, statusFailureMessage := getReasonMessageFromError(status.Failure)
// and then after "if failureReason != statusFailureReason {"
statusFailureReason=failureReason
// sand then do the same to failureMessage
// here we may avoid calling `getReasonMessageFromError(failure)` the 3rd timej by re-using `statusFailureReason` and `failureMessage`

Reply:
WDYT?
```go
	failure := status.Failure
	failingReason, failingMessage := getReasonMessageFromError(status.Failure)
	var skipFailure bool
	var progressReason, progressMessage string
	if !status.Reconciling && len(cvStatus.History) != 0 {
		progressReason, progressMessage, skipFailure = convertErrorToProgressing(now.Time, failure)
		failure = filterErrorForFailingCondition(failure, payload.UpdateEffectNone)
		filteredFailingReason, filteredFailingMessage := getReasonMessageFromError(failure)
		if failingReason != filteredFailingReason {
			klog.Infof("Filtered failure reason changed from '%s' to '%s'", failingReason, filteredFailingReason)
		}
		if failingMessage != filteredFailingMessage {
			klog.Infof("Filtered failure message changed from '%s' to '%s'", failingMessage, filteredFailingMessage)
		}
		failingReason, failingMessage = filteredFailingReason, filteredFailingMessage
	}

	// set the failing condition
	failingCondition := configv1.ClusterOperatorStatusCondition{
		Type:               ClusterStatusFailing,
		Status:             configv1.ConditionFalse,
		LastTransitionTime: now,
	}
	if failure != nil && !skipFailure {
		failingCondition.Status = configv1.ConditionTrue
		failingCondition.Reason = failingReason
		failingCondition.Message = failingMessage
	}
	resourcemerge.SetOperatorStatusCondition(&cvStatus.Conditions, failingCondition)
```

Reply:
You may check the changes in a draft commit https://github.com/openshift/cluster-version-operator/pull/1050/commits/2d5bfd2d4b788f2e142dd5819756c3cf576150b4

Reply:
LGTM